### PR TITLE
vWAN hub routing preference and autoscale

### PR DESCRIPTION
**Pull request by rozkurt**
## Overview/Summary

Added Azure vWAN Routing Preference and Virtual Hub Capacity support for Portal Deployment 

## This PR fixes/adds/changes/removes

1. Updated `vwan-connectivity.json`
2. Updated portal RI

### Breaking Changes

1. N/A

## Testing Evidence

image
image
image
image
image

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#view/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Frozkurt%2FEnterprise-Scale-rozkurt%2FvWAN_HUB_RoutingPreference_and_Autoscale%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Frozkurt%2FEnterprise-Scale-rozkurt%2FvWAN_HUB_RoutingPreference_and_Autoscale%2FeslzArm%2Feslz-portal.json)

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: eslzArm/eslz-portal.json [hunk 1]
```diff
@@ -2032,6 +2032,298 @@
...
+                            "selectAll": false,
+                            "filter": false,
+                            "multiLine": true,
+                            "visible":"[and(not(equals(steps('connectivity').enableHub, 'No')), not(equals(steps('connectivity').enableHub, 'nva')), not(equals(steps('connectivity').enableHub, 'vhub')))]",
+                            "toolTip": "Routing infrastructure units determine the minimum throughput of the Virtual WAN hub router and the number of Virtual Machines that can be deployed in Virtual Networks connected to the Virtual WAN hub. Two routing infrastructure units are included at no extra cost with a deployment of a hub.<a href=\"https://learn.microsoft.com/en-us/azure/virtual-wan/hub-settings#capacity\">Virtual Hub Capacity</a>.",
+                            "constraints": {
+                                "allowedValues": [
+                                    {
+                                        "label": "2",
+                                        "description": "2 Router Infrastructure Units, 3 Gbps Router, Supports 2000 VMs",
```
Comment (jtracey93) [inline eslzArm/eslz-portal.json:2082]:
```suggestion
                                        "description": "2 Routing Infrastructure Units, 3 Gbps Aggregate Throughput, Supports 2000 VMs",
```

Can we change all of these to match the table column names here: https://learn.microsoft.com/en-us/azure/virtual-wan/hub-settings#routing-infrastructure-unit-table

File: eslzArm/eslz-portal.json [hunk 2]
```diff
@@ -2032,6 +2032,298 @@
...
+                            "name": "vWANHubRoutingPreference",
+                            "type": "Microsoft.Common.DropDown",
+                            "label": "Hub Routing Preference",
+                            "defaultValue": "ExpressRoute",
+                            "multiselect": false,
+                            "selectAll": false,
+                            "filter": false,
+                            "multiLine": true,
+                            "visible":"[and(not(equals(steps('connectivity').enableHub, 'No')), not(equals(steps('connectivity').enableHub, 'nva')), not(equals(steps('connectivity').enableHub, 'vhub')))]",
+                            "toolTip":"Preference used in selecting best path when the virtual hub learns multiple paths to the same destination route-prefix.<a href=\"https://learn.microsoft.com/en-us/azure/virtual-wan/about-virtual-hub-routing-preference\">Virtual hub routing preference</a>.",
```
Comment (jtracey93) [inline eslzArm/eslz-portal.json:2046]:
```suggestion
                            "toolTip":"Preference used in selecting best path when the virtual hub learns multiple paths to the same destination route-prefix.<a href=\"https://learn.microsoft.com/azure/virtual-wan/about-virtual-hub-routing-preference\">Virtual hub routing preference</a>.",
```

Removing the fixed language to make this more D&I

File: docs/wiki/Whats-new.md [hunk 3]
```diff
@@ -39,6 +40,13 @@ This article will be updated as and when changes are made to the above and anyth
 
 Here's what's changed in Enterprise Scale/Azure Landing Zones:
 
+### November 2023
+
+#### Tooling
+
+- Added virtual hub routing preference support to Portal Accelerator for scenarios where you need to influence routing decisions in virtual hub router towards on-premises. For existing ALZ customers please visit [Configure virtual hub routing preference](https://learn.microsoft.com/en-us/azure/virtual-wan/howto-virtual-hub-routing-preference) for details on how to configure virtual hub routing preference settings.
+- Added virtual hub capacity option to Portal Accelerator which provides an option to select the number of routing infrastracture units. Please visit [Virtual hub capacity](https://learn.microsoft.com/en-us/azure/virtual-wan/hub-settings#capacity) for more details on Azure vWAN Virtual Hub Capacity configuration.
```
Comment (jtracey93) [inline docs/wiki/Whats-new.md:48]:
```suggestion
- Added virtual hub capacity option to Portal Accelerator which provides an option to select the number of routing infrastracture units. Please visit [Virtual hub capacity](https://learn.microsoft.com/azure/virtual-wan/hub-settings#capacity) for more details on Azure vWAN Virtual Hub Capacity configuration.
```
Removing the fixed language to make this more D&I

File: docs/wiki/Whats-new.md [hunk 4]
```diff
@@ -39,6 +40,13 @@ This article will be updated as and when changes are made to the above and anyth
 
 Here's what's changed in Enterprise Scale/Azure Landing Zones:
 
+### November 2023
+
+#### Tooling
+
+- Added virtual hub routing preference support to Portal Accelerator for scenarios where you need to influence routing decisions in virtual hub router towards on-premises. For existing ALZ customers please visit [Configure virtual hub routing preference](https://learn.microsoft.com/en-us/azure/virtual-wan/howto-virtual-hub-routing-preference) for details on how to configure virtual hub routing preference settings.
```
Comment (jtracey93) [inline docs/wiki/Whats-new.md:47]:
```suggestion
- Added virtual hub routing preference support to Portal Accelerator for scenarios where you need to influence routing decisions in virtual hub router towards on-premises. For existing ALZ customers please visit [Configure virtual hub routing preference](https://learn.microsoft.com/en-us/azure/virtual-wan/howto-virtual-hub-routing-preference) for details on how to configure virtual hub routing preference settings.
```
Removing the fixed language to make this more D&I

File: eslzArm/eslz-portal.json [hunk 5]
```diff
@@ -2032,6 +2032,298 @@
...
+                            "name": "vWANHubCapacity",
+                            "type": "Microsoft.Common.DropDown",
+                            "label": "Virtual Hub Capacity",
+                            "defaultValue": "",
+                            "multiselect": false,
+                            "selectAll": false,
+                            "filter": false,
+                            "multiLine": true,
+                            "visible":"[and(not(equals(steps('connectivity').enableHub, 'No')), not(equals(steps('connectivity').enableHub, 'nva')), not(equals(steps('connectivity').enableHub, 'vhub')))]",
+                            "toolTip": "Routing infrastructure units determine the minimum throughput of the Virtual WAN hub router and the number of Virtual Machines that can be deployed in Virtual Networks connected to the Virtual WAN hub. Two routing infrastructure units are included at no extra cost with a deployment of a hub.<a href=\"https://learn.microsoft.com/en-us/azure/virtual-wan/hub-settings#capacity\">Virtual Hub Capacity</a>.",
```
Comment (jtracey93) [inline eslzArm/eslz-portal.json:2077]:
```suggestion
                            "toolTip": "Routing infrastructure units determine the minimum throughput of the Virtual WAN hub router and the number of Virtual Machines that can be deployed in Virtual Networks connected to the Virtual WAN hub. Two routing infrastructure units are included at no extra cost with a deployment of a hub.<a href=\"https://learn.microsoft.com/azure/virtual-wan/hub-settings#capacity\">Virtual Hub Capacity</a>.",
```
Removing the fixed language to make this more D&I

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Nice work here @rozkurt 

Can we make the small changes I have suggested please

File: eslzArm/eslz-portal.json (hunk 5 above)
Comment (rozkurt) [reply eslzArm/eslz-portal.json]: @jtracey93, thanks for the comments. Just fixed them as you've suggested and pushed the changes into my branch. Could you please review again?

**Review by Springstone — APPROVED** [review]
LGTM

File: eslzArm/subscriptionTemplates/vwan-connectivity.json [hunk 6]
```diff
@@ -201,8 +222,12 @@
                                                     "id": "[variables('vwanresourceid')]"
                                                 },
                                                 "addressPrefix": "[parameters('addressPrefix')]",
-                                                "sku": "[variables('vhubsku')]"
-                                            }
+                                                "sku": "[variables('vhubsku')]",
+                                                "hubRoutingPreference": "[parameters('vWANHubRoutingPreference')]",
+                                                "virtualRouterAutoScaleConfiguration": {
+                                                "minCapacity": "[int(parameters('vWANHubCapacity'))]"
+                                                }
```
Comment (jtracey93) [inline eslzArm/subscriptionTemplates/vwan-connectivity.json:229]:
```suggestion
                                                "hubRoutingPreference": "[parameters('vWANHubRoutingPreference')]",
                                                "virtualRouterAutoScaleConfiguration": {
                                                    "minCapacity": "[int(parameters('vWANHubCapacity'))]"
                                                }
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: eslzArm/eslz-portal.json [hunk 7]
```diff
@@ -2032,6 +2032,298 @@
...
+                            "multiselect": false,
+                            "selectAll": false,
+                            "filter": false,
+                            "multiLine": true,
+                            "visible":"[and(not(equals(steps('connectivity').enableHub, 'No')), not(equals(steps('connectivity').enableHub, 'nva')), not(equals(steps('connectivity').enableHub, 'vhub')))]",
+                            "toolTip":"Preference used in selecting best path when the virtual hub learns multiple paths to the same destination route-prefix.<a href=\"https://learn.microsoft.com/azure/virtual-wan/about-virtual-hub-routing-preference\">Virtual hub routing preference</a>.",
+                            "constraints":{ 
+                                "allowedValues":[
+                                    {
+                                        "label":"ExpressRoute",
```
Comment (jtracey93) [inline eslzArm/eslz-portal.json:2050]:
```suggestion
                                        "label":"ExpressRoute (default)",
```

File: eslzArm/eslz-portal.json [hunk 8]
```diff
@@ -2032,6 +2032,298 @@
...
+                                        "description":"ExpressRoute is the preferred path",
```
Comment (jtracey93) [inline eslzArm/eslz-portal.json:2051]:
```suggestion
                                        "description":"ExpressRoute is the preferred path. (default)",
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: eslzArm/eslz-portal.json [hunk 9]
```diff
@@ -2032,6 +2032,298 @@
                                     }                                   
                                 ]
                             }
+                        },
+                        {
+                            "name": "vWANHubRoutingPreference",
+                            "type": "Microsoft.Common.DropDown",
+                            "label": "Hub Routing Preference",
+                            "defaultValue": "ExpressRoute",
```
Comment (jtracey93) [inline eslzArm/eslz-portal.json:2040]:
```suggestion
                            "defaultValue": "ExpressRoute (default)",
```

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**